Stellar-transaction.x

```diff
@@ -1790,7 +1790,9 @@ enum InvokeHostFunctionResultCode
     INVOKE_HOST_FUNCTION_TRAPPED = -2,
     INVOKE_HOST_FUNCTION_RESOURCE_LIMIT_EXCEEDED = -3,
     INVOKE_HOST_FUNCTION_ENTRY_EXPIRED = -4,
-    INVOKE_HOST_FUNCTION_INSUFFICIENT_REFUNDABLE_FEE = -5
+    INVOKE_HOST_FUNCTION_INSUFFICIENT_REFUNDABLE_FEE = -5,
+    INVOKE_HOST_FUNCTION_CODE_SIZE_ABOVE_NETWORK_LIMIT = -6,
+    INVOKE_HOST_FUNCTION_DATA_SIZE_ABOVE_NETWORK_LIMIT = -7
```

Comment:
On one hand, this is a bit risky because the apply-time errors are canonical and thus we won't be able to e.g. change order for checking the entry sizes and total read bytes or even change the fetching order between code and data entries. On the other hand, we could expand `INVOKE_HOST_FUNCTION_RESOURCE_LIMIT_EXCEEDED` as well if we're going for more granular error codes anyway. I don't have a strong opinion on this really, but introducing granular codes for just a couple of cases seems a bit weird.

Reply:
If we're going for the network limits only here, the events may exceed it as well.

Reply:
This was just a fix for the inaccurate code, but yeah this isn't consistent with the changes in TransactionFrame. I agree that adding more error codes complicates the operation (I think this was Jon's original point in not having granular error codes in Soroban). Maybe I should just use a single `INVOKE_HOST_FUNCTION_ENTRY_SIZE_ABOVE_NETWORK_LIMIT` error? Or just undo this change? Maybe we just need to document every failure case? 

Btw in addition to the events, memory is also checked against network config.

Reply:
I think the safest way would be to revert the operation change or maybe distinguish between exceeding the network limit and transaction-level limit. In any case, it would be nice to synthesize a diagnostic event that specifies what has been exceeded and how (good thing is that we don't have to do that immediately). Documentation unfortunately doesn't help debugging much (like one would know that one of 10 different things failed) and granular codes are risky. I'm also curious what @graydon and @MonsieurNicolas think.

Reply:
Yeah I'll just undo the op change and we can add diagnostic events later.